src/batch/src/task/task_execution.rs

```diff
@@ -544,22 +549,22 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
                               break;
                         },
                         None => {
-                            // TODO(ZENOTME): enable this when all executors support early stop. #9418
-                            // if self.context.get_stop_flag_ref().check_stop() {
-                            //     // Task stop by reading stop flag. It can be either cancelled or aborted.
-                            //     if let Some(error_reason) = self.context.get_stop_flag_ref().get_stop_reason() {
-                            //         state = TaskStatus::Aborted;
-                            //         error = Some(Aborted(error_reason));
-                            //     } else {
-                            //         state = TaskStatus::Cancelled;
-                            //     }
-                            // } else {
-                            //     debug!("Batch task {:?} finished successfully.", self.task_id);
-                            //     state = TaskStatus::Finished;
-                            // }
-                            debug!("Batch task {:?} finished successfully.", self.task_id);
-                            state = TaskStatus::Finished;
-                            break;
+                            match shutdown_rx.borrow().clone() {
+                                ShutdownMsg::Abort(reason) => {
+                                    state = TaskStatus::Aborted;
+                                    error = Some(Aborted(reason));
+                                    break;
+                                },
+                                ShutdownMsg::Cancel => {
+                                    state = TaskStatus::Cancelled;
+                                    break;
+                                },
+                                ShutdownMsg::Init => {
+                                    debug!("Batch task {:?} finished successfully.", self.task_id);
+                                    state = TaskStatus::Finished;
+                                    break;
+                                },
+                            }
```

Comment:
I don't think we should keep this logic here, when executor response to early drop, it will return an Err, so an `None` means a normal end.